src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/KubernetesCredentialProvider.java

```diff
@@ -107,12 +113,14 @@ public void startWatchingForSecrets() {
             LOG.log(Level.FINER, "retrieving secrets");
             SecretList list = _client.secrets().withLabelSelector(selector).withLabel(SecretUtils.JENKINS_IO_CREDENTIALS_TYPE_LABEL).list();
             ConcurrentHashMap<String, IdCredentials> _credentials = new  ConcurrentHashMap<>();
+            Multimap<String, String> _credentialScopes = Multimaps.synchronizedListMultimap(ArrayListMultimap.create());
```

Comment:
do we really need another map and guava here?
it creates threading issues and could be handled by storing a typed object in credentials containing the credential and the folder scope?

Reply:
:+1:

I didn't know this was an option, I like the suggestion.